app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt

```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
+    verify(mockDocumentReference).update(eq("activities"), any())
+  }
+
+  @Test
+  fun addLikedActivity_shouldCallFirestoreToAddLikedActivity() {
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forResult(null))
+
+    profileRepositoryFirestore.addLikedActivity("1", "activityId", onSuccess = {}, onFailure = {})
+
+    verify(mockDocumentReference).update(eq("likedActivities"), any())
+  }
+
+  @Test
+  fun removeLikedActivity_shouldCallFirestoreToRemoveLikedActivity() {
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forResult(null))
+
+    profileRepositoryFirestore.removeLikedActivity(
+        "1", "activityId", onSuccess = {}, onFailure = {})
+
+    verify(mockDocumentReference).update(eq("likedActivities"), any())
+  }
+
+  @Test
+  fun getUser_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.get()).thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.getUser(
+        "1",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
+  @Test
+  fun addActivity_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.update(eq("activities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.addActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
+  @Test
+  fun addLikedActivity_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.addLikedActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
+  @Test
+  fun removeLikedActivity_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.removeLikedActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
+  @Test
+  fun updateProfile_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.set(testUser)).thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.updateProfile(
+        testUser,
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
```

Comment:
Tests failures during adding a user profile to the database.
Validates that the failure callback handles the exception.